src/transformers/models/auto/configuration_auto.py

```diff
@@ -13,17 +13,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """ Auto Config class."""
+from __future__ import annotations
+
 import importlib
+import os
 import re
 import warnings
 from collections import OrderedDict
-from typing import List, Union
+from collections.abc import Callable, Iterator
+from typing import TYPE_CHECKING, KeysView, List, Type, TypeVar, Union, ValuesView
 
 from ...configuration_utils import PretrainedConfig
 from ...dynamic_module_utils import get_class_from_dynamic_module
 from ...utils import CONFIG_NAME, logging
 
 
+# Support Python 3.8 and below
+if TYPE_CHECKING:
+    _LazyLoadAllMappingsOrderedDict = OrderedDict[str, str]
+    _LazyConfigMappingOrderedDict = OrderedDict[str, Type[PretrainedConfig]]
+else:
+    _LazyLoadAllMappingsOrderedDict = OrderedDict
+    _LazyConfigMappingOrderedDict = OrderedDict
+
+_T = TypeVar("_T")
```

Comment:
We accept type annotations that make the code easier to read or understand. Those are not the kind that make the code easier to read.